src/libraries/System.Private.CoreLib/src/System/Double.cs

```diff
@@ -217,14 +236,21 @@ public static bool IsPositiveInfinity(double d)
             return d == PositiveInfinity;
         }
 
-        /// <summary>Determines whether the specified value is subnormal.</summary>
+        /// <summary>Determines whether the specified value is subnormal (finite, but not zero or normal).</summary>
+        /// <remarks>This effectively checks the value is not NaN, not infinite, not normal, and not zero.</remarks>
+        [NonVersionable]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsSubnormal(double d)
+        {
+            ulong bits = BitConverter.DoubleToUInt64Bits(d);
+            return ((bits & ~SignMask) - 1) < MaxTrailingSignificand;
+        }
+
         [NonVersionable]
-        // This is probably not worth inlining, it has branches and should be rarely called
-        public static unsafe bool IsSubnormal(double d)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool IsZero(double d)
```

Comment:
Do we need this helper if all it's doing is `== 0`?

Reply:
It allows the code to be consistent with `Half` (where `== 0` is less efficient today), with Generic Math (where a user really needs to use the exposed `IsZero` API), and is something we should probably expose publicly anyways since it's part of the IEEE 754 Required Operations (under 5.7.2 General Operations).